samples/matter/light_bulb/CMakeLists.txt

```diff
@@ -40,7 +40,7 @@ target_sources(app PRIVATE
     src/zap-generated/IMClusterCommandHandler.cpp
     src/zap-generated/callback-stub.cpp
     ${COMMON_ROOT}/src/led_widget.cpp
-    ${COMMON_ROOT}/src/thread_util.cpp
+    $<$<BOOL:${CONFIG_NET_L2_OPENTHREAD}>:${COMMON_ROOT}/src/thread_util.cpp>
```

Comment:
there exists a wrapper function to ease this work:
```
target_sources_ifdef(CONFIG_NET_L2_OPENTHREAD app PRIVATE ${COMMON_ROOT}/src/thread_util.cpp)
```
Ref: https://github.com/zephyrproject-rtos/zephyr/blob/2037b20db52a40d8639ea25647b362e4a4fa15e4/cmake/modules/extensions.cmake#L1515

but the current code is also acceptable, just informing of the alternative.

Reply:
Thanks for the hint, I was not aware of that.